src/queryParser.js

```diff
@@ -6,15 +6,15 @@ export function parseQuery (query) {
   }
   if (query.orderBy || query.sort) {
     const orderBy = query.orderBy || query.sort
-    if (orderBy.length) {
+    if (Array.isArray(orderBy)) {
       query.orderBy = orderBy.map((clause) => {
-        if (typeof clause === 'string') {
+        if (typeof clause === 'string' && clause.indexOf('{') >= 0) {
           return JSON.parse(clause)
         }
         return clause
       })
-      query.sort = undefined
     }
+    query.sort = undefined
```

Comment:
@eljenso You do not want to assign `query.sort=undefined` here, it should remain inside the block where `oderBy` is defined.

Reply:
@crobinson42 This is actually the same block as where `const orderBy = query.orderBy || query.sort` is.